rust_src/src/data.rs

```diff
@@ -346,6 +348,70 @@ pub fn byteorder() -> u8 {
     }
 }
 
+/// Return the default value of SYMBOL, but don't check for voidness.
+/// Return Qunbound if it is void.
+fn default_value(mut symbol: LispSymbolRef) -> LispObject {
+    while symbol.get_redirect() == symbol_redirect::SYMBOL_VARALIAS {
+        symbol = symbol.get_indirect_variable();
+    }
+    match symbol.get_redirect() {
+        symbol_redirect::SYMBOL_PLAINVAL => symbol.get_value(),
+        symbol_redirect::SYMBOL_LOCALIZED => {
+            // If var is set up for a buffer that lacks a local value for it,
+            // the current value is nominally the default value.
+            // But the `realvalue' slot may be more up to date, since
+            // ordinary setq stores just that slot.  So use that.
+            let blv = symbol.get_blv();
+            let fwd = blv.get_fwd();
+            if !fwd.is_null() && blv.valcell.eq(blv.defcell) {
+                unsafe { do_symval_forwarding(fwd) }
+            } else {
+                blv.defcell.as_cons_or_error().cdr()
+            }
+        }
+        symbol_redirect::SYMBOL_FORWARDED => {
+            let valcontents = symbol.get_fwd();
+
```

Comment:
I am leaning negative on this use of `if_chain`. I think in part because of the `let` statement. But also because it causes the reader to stop and process more. I have to question each line. Did I read it right?

Reply:
A guide comment has been put below to summarize what `if_chain!` does and note that it should be removed when we're off of raw pointer dereferencing and onto Option. The main reasons I went with if_chain here were that an early return inside a `match`, `if`, and `if` would probably cause the reader to check what it's missing (here, nothing, but a full function skim is likely) due to the amount of nesting, and this should be easier to convert into combinators later that also won't be early returning when higher Rust coverage arrives. I'd also bet that there are some other similarly nasty control flow bits elsewhere in Remacs, and having this at hand could help with clearly porting those.